File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/commandbase/InstantCommand.java
package org.firstinspires.ftc.teamcode.common.commandbase;

/**
 * Usage:
 * new InstantCommand(() -> {class.method();});
 */
public class InstantCommand implements Command
{
	private VoidFunction voidFunction;

	public InstantCommand (VoidFunction voidFunction)
	{
		this.voidFunction = voidFunction;
	}

	@Override
	public boolean run()
	{
		voidFunction.run();
		return true;
	}
}

File: TeamCode/src/main/java/brickbot/quickstart/commandbase/InstantCommand.kt
package brickbot.quickstart.commandbase

class InstantCommand(
    commandName: String = "",
    private val unitFunction: Runnable
): Command(commandName) {

    override fun run(): Boolean {
        unitFunction.run()
        return true;
    }

    override fun clone(): Command {
        return InstantCommand(commandName, unitFunction)
    }
}